patches/tModLoader/Terraria/GameContent/UI/ResourceSets/PlayerResourceSetsManager.cs.patch

```diff
@@ -0,0 +1,48 @@
+--- src/TerrariaNetCore/Terraria/GameContent/UI/ResourceSets/PlayerResourceSetsManager.cs
++++ src/tModLoader/Terraria/GameContent/UI/ResourceSets/PlayerResourceSetsManager.cs
+@@ -5,7 +_,7 @@
+ 
+ namespace Terraria.GameContent.UI.ResourceSets
+ {
+-	public class PlayerResourceSetsManager
++	public partial class PlayerResourceSetsManager
+ 	{
+ 		private Dictionary<string, IPlayerResourcesDisplaySet> _sets = new Dictionary<string, IPlayerResourcesDisplaySet>();
+ 		private IPlayerResourcesDisplaySet _activeSet;
+@@ -23,7 +_,7 @@
+ 		}
+ 
+ 		private void Configuration_OnLoad(Preferences obj) {
+-			_activeSetConfigKey = Main.Configuration.Get("PlayerResourcesSet", "New");
++			_activeSetConfigKeyOriginal = _activeSetConfigKey = Main.Configuration.Get("PlayerResourcesSet", "New");
+ 			if (_loadedContent)
+ 				SetActiveFromLoadedConfigKey();
+ 		}
+@@ -35,8 +_,8 @@
+ 
+ 		public void LoadContent(AssetRequestMode mode) {
+ 			_sets["New"] = new FancyClassicPlayerResourcesDisplaySet("New", "New", "FancyClassic", mode);
+-			_sets["Default"] = new ClassicPlayerResourcesDisplaySet("Default", "Default");
+ 			_sets["HorizontalBars"] = new HorizontalBarsPlayerReosurcesDisplaySet("HorizontalBars", "HorizontalBars", "HorizontalBars", mode);
++			_sets["Default"] = new ClassicPlayerResourcesDisplaySet("Default", "Default");
```

Comment:
why re-order?

Reply:
The values of a dictionary aren't ordered anyway.

Reply:
they're in the ordered in the order they're added at when using FirstOrDefault

Reply:
Well I looked at the reference source, and apparently it's true for this version of .NET

I don't like relying on it though, it's confusing to other developers who follow the typical advice (that dictionaries and hashmaps are unordered)

From MSDN
![image](https://user-images.githubusercontent.com/388233/191880518-7efba9ce-cde0-4adb-8a88-725b97636dcf.png)

Reply:
fair point, though that's what the vanilla logic uses.  though i should mention that `Dictionary<K, V>.Values` is what's being iterated

Reply:
Yeah, the version of `.NET` we're using stores all the entries in an `Entry[]` in insertion order (at least when you don't remove and re-add stuff)